stdlib/knative/src/main/ballerina/src/knative/annotation.bal

```diff
@@ -0,0 +1,232 @@
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+# Knative annotation configuration.
+#
+#
+# # metadata for artifacts
+# Metadata for artifacts
+#
+# + name - Name of the resource. Default is `"<OUPUT_FILE_NAME>-<RESOURCE_TYPE>"`.
+# + labels - Map of labels for the resource.
+# + annotations - Map of annotations for resource.
+public type Metadata record {|
+    string name?;
+    map<string> labels?;
+    map<string> annotations?;
+|};
+# External file type for docker.
+#
+# + sourceFile - source path of the file (in your machine)
+# + target - target path (inside container)
+public type FileConfig record {|
+    string sourceFile;
+    string target;
+|};
+# Value for a field.
+#
+# + fieldPath - Path of the field
+public type FieldValue record{|
+    string fieldPath;
+|};
+# Value for a secret key.
+#
+# + name - Name of the secret.
+# + key - Key of the secret.
+public type SecretKeyValue record {|
+    string name;
+    string key;
+|};
+# Value for config map key.
+#
+# + name - name of the config.
+# + key - key of the config.
+public type ConfigMapKeyValue record {|
+    string name;
+    string key;
+|};
+# Value for resource field.
+#
+# + containerName - Name of the container.
+# + resource - Resource field
+public type ResourceFieldValue record {|
+   string containerName?;
+   string 'resource;
+|};
+# Value from config map key.
+#
+# + configMapKeyRef - Reference for config map key.
+public type ConfigMapKeyRef record {|
+    ConfigMapKeyValue configMapKeyRef;
+|};
+# Value from field.
+#
+# + fieldRef - Reference for a field.
+public type FieldRef record {|
+    FieldValue fieldRef;
+|};
+# Value from secret key.
+#
+# + secretKeyRef - Reference for secret key.
+public type SecretKeyRef record {|
+    SecretKeyValue secretKeyRef;
+|};
+# Value from resource field.
+#
+# + resourceFieldRef - Reference for resource field.
+public type ResourceFieldRef record {|
+    ResourceFieldValue resourceFieldRef;
+|};
+public const string IMAGE_PULL_POLICY_IF_NOT_PRESENT = "IfNotPresent";
+public const string IMAGE_PULL_POLICY_ALWAYS = "Always";
+public const string IMAGE_PULL_POLICY_NEVER = "Never";
+# Image pull policy type field for kubernetes deployment and jobs.
+public type ImagePullPolicy "IfNotPresent"|"Always"|"Never";
+# Probing configuration.
+#
+# + port - Port to check for tcp connection.
+# + initialDelaySeconds - Initial delay for pobing in seconds.
+# + periodSeconds - Interval between probes in seconds.
+public type ProbeConfiguration record {|
+    int port?;
+    int initialDelaySeconds?;
+    int periodSeconds?;
+|};
+# Type of operations between key and value of a toleration.
+public type TolerationOperator "Exists"|"Equal";
+# Types of toleration effects for pods.
+public type TolerationEffect "NoSchedule"|"PreferNoSchedule"|"NoExecute";
```

Comment:
Shall we introduce constants for these too?

Reply:
constants added. please check